File: src/main/java/fundamentals/java/collections/JavaArray.java
package fundamentals.java.collections;

public class JavaArray {
    public static void main(String[] args) {
        int[] arr = new int[5];
        for (int anArr : arr) System.out.println(anArr);
    }
}

class ExplicitArrayInitialization {
    public static void main(String[] args) {
        // Initialize the array at the time of declaration
        int[] arr = {1, 2, 3, 4, 5};

        int[] arr1 = {1, 2, 3, 4, 5, }; // A comma after the last value 5 is valid.

        int[] arr2 = new int[]{1, 2, 3, 4, 5};

        int[] empty = { };

        for (int i : arr) System.out.println(arr[i]);
    }
}

File: src/main/scala/fundamentals/scala/collections/JavaArray.scala
package fundamentals.scala.collections

/**
 * Array is a special kind of collection in Scala. On the one hand, Scala arrays correspond
 * one-to-one to Java arrays. That is, a Scala array Array[Int] is represented as a Java int[],
 * an Array[Double] is represented as a Java double[] and a Array[String] is represented as a
 * Java String[]. But at the same time, Scala arrays offer much more than their Java analogues.
 * First, Scala arrays can be generic.
 */
object JavaArray extends App {
  val arr = new Array[Int](5)
  for (i <- arr) println(i)
}

object ExplicitArrayInitialization extends App {
  val arr = Array(1, 2, 3, 4, 5)

  val empty1: Array[Int] = Array()

  val empty2: Array[Int] = Array.empty

  for (i <- arr) println(i)
}
